cmd/depdiff/dependencies.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package depdiff
+
+// ChangeType is the change type (added, updated, removed) of a dependency.
+type ChangeType string
+
+const (
+	Added   ChangeType = "added"
+	Updated ChangeType = "updated"
+	Removed ChangeType = "removed"
+)
+
+// IsValid determines if a ChangeType is valid.
+func (ct *ChangeType) IsValid() bool {
+	switch *ct {
+	case Added, Updated, Removed:
+		return true
+	default:
+		return false
+	}
+}
+
+// Dependency is a dependency.
+type Dependency struct {
+	// // IsDirect suggests if the dependency is a direct dependency of a code commit.
+	// TODO: IsDirect remains a future feature since the current GitHub Dependency Review API
+	// mixes up direct and indirect dependencies in manifest files of different ecosystems.
+	IsDirect bool
+
+	// ChangeType indicates whether the dependency is added or removed.
+	ChangeType ChangeType `json:"change_type"`
+
+	// ManifestFileName is the name of the manifest file of the dependency, such as go.mod for Go.
+	ManifestFileName string `json:"manifest"`
+
+	// Ecosystem is the name of the package management system, such as NPM, GO, PYPI.
+	Ecosystem string `json:"ecosystem" bigquery:"System"`
+
+	// Name is the name of the dependency.
+	Name string `json:"name" bigquery:"Name"`
```

Comment:
Maybe make all these fields pointers, because in some cases they won't be populated?

Reply:
Update: since the `License` field is duplicated and has been removed from the `Dependency` struct, only `package_url` and `source_repository_url` are changed to pointers for now.

Reply:
The structure should be generic and not depend on the GH API. Right now we populate it with the GH API but this may change in the future.

Reply:
Thank you so much